package_buildpack.go

```diff
@@ -144,9 +132,11 @@ func (c *Client) PackageBuildpack(ctx context.Context, opts PackageBuildpackOpti
 
 	switch opts.Format {
 	case FormatFile:
-		return packageBuilder.SaveAsFile(opts.Name, opts.OS)
+		return packageBuilder.SaveAsFile(opts.Name, opts.Config.Platform.OS)
 	case FormatImage:
-		_, err = packageBuilder.SaveAsImage(opts.Name, opts.Publish, opts.OS)
+		//TODO: should check if saving linux image on windows daemon ???
```

Comment:
It would fail already, but if you wanted a friendlier message then checking here would be a good route